src/System.IO.FileSystem/src/System/IO/CharSpanExtensions.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.IO
+{
+    internal static partial class CharSpanExtensions
+    {
+        public static bool EndsWithOrdinal(this ReadOnlySpan<char> span, ReadOnlySpan<char> value, bool ignoreCase = false)
+        {
+            if (value.Length == 0)
+                return true;
+            else if (value.Length > span.Length)
+                return false;
+
+            if (span.Length != value.Length)
+                span = span.Slice(span.Length - value.Length);
```

Comment:
How common is it for them to have the same length?  I'm wondering if this should just be simplified to remove the if and just always slice.  Just curious.

Reply:
It will happen, say looking for "*readme.txt", but not common. I assumed the check was cheap enough, but I can remove it.